docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,193 @@
+
+[#supported-languages]
+.Supported languages
+
+[cols="1,1"]
+|===
+|Language
+|Package Manager
+
+|Go
+|`gomod`
+
+|Python
+|`pip`
+|===
+
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a `go.mod` file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: gomod <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `gomod`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+include::../partials/con_hermetic_verification.adoc[]
+
+== Enabling prefetch builds for `pip`
+Cachi2 supports pip by parsing of `pip` requirements files, including but not limited to *requirements.txt* files placed in the root of your repository. By generically parsing `pip` requirements files, it downloads the specified dependencies.
+
+IMPORTANT: The requirements file can have a different name, as you can use multiple files to provide the dependencies. These requirements files function as lockfiles, encompassing all the transitive dependencies. It's crucial to actively pin each transitive dependency listed in the requirements file to a specific version.
+
+
+s, which include all the transitive dependencies. Each transitive dependency in the *requirements.txt* file must be pinned to a specific version.
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a <filename> file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component’s source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: pip <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `pip`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in the JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+. Create a file (for example, pyproject.toml) with metadata (name and version) and place it in the root of your repository.
```

Comment:
If your project is a real python project, then a metadata file will already exist. It's good to explain where cachi2 will look for the project metadata, but the wording may use some changes

Note: cachi2 currently does _require_ a metadata file, but we want to change that as soon as our intern gets back :smile:

Reply:
@chmeliik Do you agree we can update line # 77 to:

Review the root of your repository for a metadata file, for example, *pyproject.toml* file, *setup.py*, or *setup.cfg*. If there is no metadata file, create one, as Cachi2 looks for the name and version of your project in the metadata files.

Reply:
BTW, I've checked in this change, let me know if we are good.

Reply:
Not necessarily the root of the repository; keep in mind that the user can specify the `path` to the package (by default, the path is `.`, the repo root). The metadata file should be at this path.

Otherwise sounds good to me